src/ifd-nfc.c

```diff
@@ -102,6 +103,8 @@ void log_msg(const int priority, const char *fmt, ...)
 #include <inttypes.h>
 #include <time.h>
 
+extern char *strdup(char *);
```

Comment:
To my surprise it turns out that strdup() is not POSIX and compiler is flagged for POSIX.. hence even when I included string.h there was no prototype for strdup().  I didn't want to remove POSIX compliance because I didn't know what platforms this code is being used on.. so I replaced my strdup with malloc+strcpy.  This line of code is a leftover, I will remove it.